packages/vaadin-virtualizer/src/iron-list.js

```diff
@@ -441,29 +464,39 @@ export const PolymerIronList = Class({
    * Increases the pool size.
    */
   _increasePoolIfNeeded: function (count) {
-    const nextPhysicalCount = this._clamp(
+    var nextPhysicalCount = this._clamp(
       this._physicalCount + count,
       DEFAULT_PHYSICAL_COUNT,
       this._virtualCount - this._virtualStart
     );
-    const delta = nextPhysicalCount - this._physicalCount;
-    let nextIncrease = Math.round(this._physicalCount * 0.5);
+    nextPhysicalCount = this._convertIndexToCompleteRow(nextPhysicalCount);
+    if (this.grid) {
```

Comment:
We can probably remove the code related to `grid` mode.

Reply:
Can be removed later once we have a conclusion for https://vaadin.slack.com/archives/G01DE1TBFFX/p1619608299042300